File: app/src/main/java/com/example/parttimecalander/SplashActivity.java
package com.example.parttimecalander;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;

import androidx.appcompat.app.AppCompatActivity;

import com.example.parttimecalander.home.HomeActivity;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        // Simulate data loading using a handler and Runnable (as Kotlin's coroutine is replaced by this in Java)
        new Handler(Looper.getMainLooper()).postDelayed(() -> {
            // Preload data and go to the next activity after the delay
            preloadMainActivityData();
            goToHomeActivity();
        }, 200); // Delay time in milliseconds
    }

    private void preloadMainActivityData() {
        // This method would normally load data from a database, etc.
        // Simulated delay
        try {
            Thread.sleep(1000); // Simulating delay
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    private void goToHomeActivity() {
        Intent intent = new Intent(SplashActivity.this, HomeActivity.class);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/example/parttimecalander/SplashActivity.kt
package com.example.parttimecalander

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import kotlinx.coroutines.*

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        lifecycleScope.launch{
            preloadMainActivityData()
            goToMainActivity()
        }
    }

    private suspend fun  preloadMainActivityData(){
        withContext(Dispatchers.IO){
            //TODO: pre_load mainActivity data from database
            delay(2000)
        }
    }

    private fun goToMainActivity(){
        val intent = Intent(this@SplashActivity, MainActivity::class.java)
        startActivity(intent)
        finish()

    }
}